README.md

```diff
@@ -129,6 +128,28 @@ You can read more about what each of these objects is for on the [following link
 - The write doesn't validate the structure of the project. It's up to the developer to validate the changes that have been done using `xcproj`.
 - New versions of Xcode might introduce new models or property that are not supported by `xcproj`. If you find any, don't hesitate to [open an issue](https://github.com/xcodeswift/xcproj/issues/new) on the repository.
 
+## Examples
+
+### Creating a group inside another group
+
+```swift
+let project = try! XcodeProj(path: "myproject.xcodeproj")
+let myGroup = PBXGroup(reference: "xxx", children: [], sourceTree: .group, path: "MyGroup")
+project.pbxproj.append(myGroup)
```

Comment:
This wouldn't actually make the group visible. You should probably use the reference generator as well.
Also is `pbxproj.append` a valid function?

Reply:
It should be `project.pbxproj.objects.append(myGroup)`. What does it need to be visible? Adding it to another group?
Regarding the reference generator, since it's not merged yet, I'd say leave it like that until we have something in place to handle them. what do you think?

Reply:
I've updated the examples. What do you think @yonaskolb ?